docs/roman/associations/skycell_asn.rst

```diff
@@ -0,0 +1,91 @@
+.. _skycell_asn:
+
+skycell_asn
+===========
+
+Create an association using either the command line tool
+``skycell_asn`` or through the Python API using either
+:class:`romancal.associations.skycell_asn.Main` or
+:func:`romancal.associations.skycellasn.skycell_asn`
+
+
+Associations
+^^^^^^^^^^^^
+
+Refer to TBD for a full description of associations.
+
+To create an association, use the following command:
+
+.. code-block:: python
+
+		skycell_asn file_list -o my_program
+
+where the individual calibrated detector files are in the current directory and the -o is the root
+for the output associations. To get a complete list of options you can run the command with the
+\-h option
+
+.. code-block:: python
+
+		skycell_asn -h
+
+
+By knowing the structure of the conventional file name you can generate custom associations based
+on the information included in the file names.
+The current naming convention for the input files consists of a Visit_ID + Exposure_ID + detector + suffix
+Where the Visit_ID is a 19 digit string
+PPPPPCCAAASSSOOOVVV, with
+
+PPPPP = Program number
+CC = Execution plan number
+AAA = Pass number (within execution plan)
+SSS = Segment Number (within pass)
+OOO = Observation number (within the leg, not to be confused with the Observation ID)
+VVV = Visit Number (within the observation)
+
+and the Exposure_ID is a four digits designating the exposures
+
+eeee = Exposure number (within the visit)
+
+The detector is WFI01, WFI02, ... WFI18
+
+The suffix indicates the processing level of the file and to create products based on the
+skycells the suffix should generally be 'cal'.
+
+and the file name is constructed as,
+rPPPPPCCAAASSSOOOVVV_eeee_detector_suffix.asdf
+
+To give a more concrete example we'll use the APT example program for the
+High Latitude Wide Angle Survey (hlwas). The program is 00991, the execution plan number is 01,
+there are 12 passes, 5 segments, 4 observations and a various number of visits.
+For this example we want to select a single filter, say F158, and that is segment 003.
```

Comment:
I'm not sure I follow this example.  When I look at program 991 and pass 1 segment 3, I see a lot of exposures in different filters.  Specifically, each visit is a linegap 4 dither, and those are grouped into observations of 32 pointings in a mosaic, and those observations are grouped into a segment with the F106, F129, F158, F184, and GRISM elements.

So I think the HLWAS example program has:
- 12 passes (but note there are really only 2 passes of imaging, which are the ones we care about and the other passes are purely spectroscopic)
- 169 segments per spectroscopic pass (basically, a big mosaic covering a lot of the sky)
- 5 observations per segment (F106, F129, F158, F184, GRISM)
- 32 visits per observation (a smaller mosaic covering a patch)
- 4 exposures per visit (linegap 4 dither)
So to select out only F158 I think you need to get observation 3.

It does look to me like it's required that all exposures in the same "observation" use the same filter, which I hadn't appreciated before.

Reply:
I've updated the numbers, I was trying to simplify this for the docs. Is there another program you know of that might be clearer?

Reply:
The HLWAS makes sense because it's one people will have heard of.  There are much smaller programs that one could use but then I just think they will be unknown to people.  In general, I am not sure how well publicized the example APT programs are, but I did think it was useful to have a "worked example"; this was a reminder for me that the "execution plan" stuff is a weird detail that is pretty extraneous to our planning.  It's possible we would want to ?? over that as well, but then I don't know what base name I would use, so I didn't suggest anything there.

Reply:
The new version still selects segment 3, but I think you need observation 3.  i.e., the names need to be more like r0099101001001003001_*_cal.asdf, likewise the base names.